components/match2/wikis/teamfortress/match_legacy.lua

```diff
@@ -0,0 +1,130 @@
+--
+
+local MatchLegacy = {}
+
+local Json = require('Module:Json')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper')
+
+function MatchLegacy.storeMatch(match2, options)
+	local match = MatchLegacy._convertParameters(match2)
+
+	if options.storeMatch1 then
+		match.games = MatchLegacy.storeGames(match, match2)
+
+		return mw.ext.LiquipediaDB.lpdb_match(
+			'legacymatch_' .. match2.match2id,
+			match
+		)
+	end
+end
+
+function MatchLegacy.storeGames(match, match2)
+	local games = ''
+	for gameIndex, game2 in ipairs(match2.match2games or {}) do
+		local game = Table.deepCopy(game2)
+		-- Extradata
+		game.extradata = {}
+		game.extradata.gamenumber = gameIndex
+		-- Other stuff
+		game.opponent1 = match.opponent1
+		game.opponent2 = match.opponent2
+		game.opponent1flag = match.opponent1flag
+		game.opponent2flag = match.opponent2flag
+		game.date = match.date
+		local scores = game2.scores or {}
+		if type(scores) == 'string' then
+			scores = Json.parse(scores)
+		end
+		game.opponent1score = scores[1] or 0
+		game.opponent2score = scores[2] or 0
+		local res = mw.ext.LiquipediaDB.lpdb_game(
+			'legacygame_' .. match2.match2id .. gameIndex,
+			Json.stringifySubTables(game)
+		)
+		games = games .. res
+	end
+	return games
+end
+
+function MatchLegacy._convertParameters(match2)
+	local match = Table.deepCopy(match2)
+	for key, _ in pairs(match) do
+		if String.startsWith(key, 'match2') then
+			match[key] = nil
+		end
+	end
+
+	if match.walkover == 'ff' or match.walkover == 'dq' then
+		match.walkover = match.winner
+	elseif match.walkover == 'l' then
+		match.walkover = nil
+	end
+
+	match.staticid = match2.match2id
+
+	-- Handle extradata fields
+	match.extradata = {}
+	local extradata = Json.parseIfString(match2.extradata)
+
+	local mvp = Json.parseIfString(extradata.mvp)
+	if mvp and mvp.players then
+		local players = {}
+		for _, player in ipairs(mvp.players) do
+			table.insert(players, player.name .. '|' .. player.displayname)
+		end
+		match.extradata.mvp = table.concat(players, ',')
+		match.extradata.mvp = match.extradata.mvp .. ';' .. mvp.points
+	end
+
+	match.extradata.matchsection = extradata.matchsection
+	match.extradata.bestofx = match2.bestof ~= 0 and tostring(match2.bestof) or ''
+	local bracketData = Json.parseIfString(match2.match2bracketdata)
+	if type(bracketData) == 'table' and bracketData.type == 'bracket' then
+		if bracketData.inheritedheader then
+			match.header = (DisplayHelper.expandHeader(bracketData.inheritedheader) or {})[1]
+		end
+	end
+
+	-- Handle Opponents
+	local handleOpponent = function (index)
+		local prefix = 'opponent'..index
+		local opponent = match2.match2opponents[index] or {}
+		local opponentmatch2players = opponent.match2players or {}
+		if opponent.type == 'team' then
+			match[prefix] = mw.ext.TeamTemplate.teampage(opponent.template)
```

Comment:
This is an AoE specific thing afaik, does it need to be here, or can the usual one be used?

Reply:
replacing this with `match[prefix] = opponent.name` now 
or that line should just be kicked in general?